data/synth_timing.xdc

```diff
@@ -292,20 +296,58 @@ set usb_conf_names {usrusb_softcn usrusb_sus usrusb_spd}
 set_output_delay -clock vclk_extusb -max 0 [get_ports $usb_conf_names]
 set_output_delay -clock vclk_extusb -min 0 [get_ports $usb_conf_names]
 
-## PMOD 0 - asynchronous GPIO
+## PMOD 0
+#   SPI (or I^2C or UART or GPIO, but let us assume SPI is most critical)
+# Use *similar* methodology as Arduino Shield SPI, but with less I/O delay on
+# some shared ports to avoid timing violations on I^2C and UART paths.
+# Distance to PMOD 0 SPI pins = ~20-25 mm  x2(clk there + data back)
+set pmod0_spi_trce_dly_min [expr {20 * 2 * 0.004}]
+set_input_delay -clock clk_pmod0_spi -max [expr {$sclk_ns - 22                         }] [get_ports {pmod0[2]}] ;# CIPO
```

Comment:
Why is this minus 22. In the Arduino it is minus 16.

Reply:
This is relaxed a bit compared to the Arduino SPI to make the I^2C paths happy. Supposedly documented in the comment above, though perhaps that isn't clear.